components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -284,53 +295,92 @@ public String getUsernameByClaims(Map<String, String> claims, String tenantDomai
                     null);
         }
         int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-        String[] resultedUserList = ArrayUtils.EMPTY_STRING_ARRAY;
-        for (String key : claims.keySet()) {
-            String value = claims.get(key);
-            if (StringUtils.isNotEmpty(key) && StringUtils.isNotEmpty(value)) {
-                if (log.isDebugEnabled()) {
-                    String message = String.format("Searching users by claim : %s for value : %s", key, value);
-                    log.debug(message);
+        try {
+            String[] userStoreDomainNames = getDomainNames(tenantId);
+            carbonUM = (AbstractUserStoreManager) getUserStoreManager(tenantId);
+            RealmService realmService = IdentityRecoveryServiceDataHolder.getInstance().getRealmService();
+            ClaimManager claimManager = (ClaimManager) realmService.getTenantUserRealm(IdentityTenantUtil.
+                    getTenantId(tenantDomain)).getClaimManager();
+            ArrayList<org.wso2.carbon.user.core.common.User> resultedUserList = new ArrayList<>();
+
+            for (String domain : userStoreDomainNames) {
+                List<ExpressionCondition> expressionConditionList =
+                        getExpressionConditionList(claims, domain, claimManager);
+
+                if (!expressionConditionList.isEmpty()) {
+                    Condition operationalCondition = getOperationalCondition(expressionConditionList);
+                    resultedUserList.addAll(carbonUM.getUserListWithID(operationalCondition, domain,
+                            UserCoreConstants.DEFAULT_PROFILE, 2, 1, null, null));
```

Comment:
Shall we a simple comment mentioning what is passed here as 2,1,null,null

Reply:
Resolved by https://github.com/wso2-extensions/identity-governance/pull/623/commits/ec87914bdfabae3c2012cda51e59249fadeefe94